plasmapy/classes/sources/tests/test_openpmd_hdf5.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import plasmapy
+from astropy import units as u
+import os
+
+@pytest.fixture(scope="module")
+def openPMD2DPlasma():
+    """This is me not knowing how to handle the new Plasma class with data!
+
+    Downloaded from https://github.com/openPMD/openPMD-example-datasets/blob/draft/example-2d.tar.gz
+    per the Creative Commons Zero v1.0 Universal license"""
+    return plasmapy.classes.Plasma(
+        hdf5=os.path.join(
+            os.path.dirname(__file__) or '.',
+            "data",
+            "data00000255.h5")
+        )
+
+def test_has_electric_field_with_units(openPMD2DPlasma):
+    "It would be nice to let the fields have the correct units from the get go"
+    assert openPMD2DPlasma.electric_field.to(u.V / u.m)
+
+def test_has_charge_density_with_units(openPMD2DPlasma):
+    assert openPMD2DPlasma.charge_density.to(u.C/u.m**3)  # unless it's some
+    # 2D charge density
+
+def test_correct_shape_electric_field(openPMD2DPlasma):
+    assert openPMD2DPlasma.electric_field.shape == (3, 51, 201)
+    # IIRC this is how we defined it in the old Plasma class but I can be
+    # convinced otherwise if another way makes more sense
+
+def test_x(openPMD2DPlasma):
+    assert openPMD2DPlasma.x.shape == (51, 201)
```

Comment:
This is whose position coordinates we are talking about? If this is for particles like electrons, etc., it would make more sense to have something like `openPMD2DPlasma.electrons.x`.

Reply:
It definitely would! I thought I mentioned it, but guess not. OpenPMD lets you attach discrete particle data to your mesh data, but I think it would be fine to skip the `particle`s for now and just handle the mesh data well. Adding particles would mean we'd have to rethink our `species` class... which we could, of course, do.

Also, I now have doubts about the fact that OpenPMD lets you define separate coordinates for each field (e.g. you can have electric and magnetic fields defined at a spatial offset to each other, like for [Yee grids](https://upload.wikimedia.org/wikipedia/commons/thumb/1/1b/FDTD_Yee_grid_2d-3d.svg/800px-FDTD_Yee_grid_2d-3d.svg.png) )... we'll have to think more about how to handle this...

Reply:
> I think it would be fine to skip the particles for now

yes, fully optional. mesh-only works as well.

> [o]penPMD lets you define separate coordinates for each field

Just in case this is relevant for you. Otherwise just assign them to zero for now if it's irrelevant for your use case (e.g. node-centered).